src/rdkafka_metadata.c

```diff
@@ -508,6 +508,20 @@ rd_kafka_parse_Metadata (rd_kafka_broker_t *rkb,
         rd_kafka_wrlock(rkb->rkb_rk);
         rkb->rkb_rk->rk_ts_metadata = rd_clock();
 
+        /* Update cached cluster id. */
+        if (RD_KAFKAP_STR_LEN(&cluster_id) > 0 &&
+            (!rkb->rkb_rk->rk_clusterid ||
+             rd_kafkap_str_cmp_str(&cluster_id, rkb->rkb_rk->rk_clusterid))) {
+                rd_rkb_dbg(rkb, BROKER|RD_KAFKA_DBG_GENERIC, "CLUSTERID",
```

Comment:
Sounds like this was something punted on for the first version of cluster ID, so I guess matching the Java behavior will make sense. The idea is that if you really want to make sure you're working against a specific cluster, it's something you could specify via config and then validate here rather than looking for changes in the ID.